test/Libraries/Microsoft.Extensions.Caching.Hybrid.Tests/SizeTests.cs

```diff
@@ -0,0 +1,95 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.Extensions.Caching.Hybrid.Internal;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Microsoft.Extensions.Caching.Hybrid.Tests;
+
+public class SizeTests
+{
+    [Theory]
+    [InlineData(null, true)] // does not enforce size limits
+    [InlineData(8L, false)] // unreasonably small limit; chosen because our test string has length 12 - hence no expectation to find the second time
+    [InlineData(1024L, true)] // reasonable size limit
+    public async Task ValidateSizeLimit_Immutable(long? sizeLimit, bool expectFromL1)
+    {
+        var services = new ServiceCollection();
+        services.AddMemoryCache(options => options.SizeLimit = sizeLimit);
+        services.AddHybridCache();
+        using var provider = services.BuildServiceProvider();
```

Comment:
ditto

Reply:
addressed in [5e3bdba](https://github.com/dotnet/extensions/pull/5420/commits/5e3bdbafb8f2b59c811c7d66fb225a11ef379e2a)